src/icrar/leap-accelerate/algorithm/casa/PhaseRotate.cc

```diff
@@ -143,37 +198,40 @@ namespace casalib
-        // {
-        //     // For baseline
-        //     const double pi = boost::math::constants::pi<double>();
-        //     double shiftFactor = -2 * pi * uvw[baseline].get()[2] - metadata.oldUVW[baseline].get()[2]; // check these are correct
-        //     shiftFactor = shiftFactor + 2 * pi * (metadata.phase_centre_ra_rad * metadata.oldUVW[baseline].get()[0]);
-        //     shiftFactor = shiftFactor - 2 * pi * (direction.get()[0] * uvw[baseline].get()[0] - direction.get()[1] * uvw[baseline].get()[1]);
-
-        //     if(baseline % 1000 == 1)
-        //     {
-        //         std::cout << "ShiftFactor for baseline " << baseline << " is " << shiftFactor << std::endl;
-        //     }
-
-        //     // Loop over channels
-        //     for(int channel = 0; channel < metadata.channels; channel++)
-        //     {
-        //         double shiftRad = shiftFactor / metadata.channel_wavelength[channel];
-
-        //         Eigen::VectorXcd v = data(channel, baseline);
-        //         data(channel, baseline) = v * std::exp(std::complex<double>(0.0, 1.0) * std::complex<double>(shiftRad, 0.0));
-
-        //         if(!data(channel, baseline).hasNaN())
-        //         {
-        //             for(int i = 0; i < data(channel, baseline).cols(); i++)
-        //             {
-        //                 metadata.avg_data.get()(baseline, i) += data(channel, baseline)(i);
-        //             }
-        //         }
-        //     }
-        // }
+        for(int baseline = 0; baseline < integration.baselines; ++baseline)
+        {
+            // For baseline
+            const double pi = boost::math::constants::pi<double>();
+            double shiftFactor = -2 * pi * uvw[baseline].get()[2] - metadata.oldUVW[baseline].get()[2]; // check these are correct
+
+            shiftFactor = shiftFactor + 2 * pi * (metadata.phase_centre_ra_rad * metadata.oldUVW[baseline].get()[0]);
+            shiftFactor = shiftFactor - 2 * pi * (direction.get()[0] * uvw[baseline].get()[0] - direction.get()[1] * uvw[baseline].get()[1]);
+
+            if(baseline % 1000 == 1)
+            {
+                std::cout << "ShiftFactor for baseline " << baseline << " is " << shiftFactor << std::endl;
+            }
+
+            // Loop over channels
+            for(int channel = 0; channel < metadata.channels; channel++)
+            {
+                double shiftRad = shiftFactor / metadata.channel_wavelength[channel];
+
+                Eigen::VectorXcd v = data(channel, baseline);
+                data(channel, baseline) = v * std::exp(std::complex<double>(0.0, 1.0) * std::complex<double>(shiftRad, 0.0));
+
+                if(!data(channel, baseline).hasNaN())
+                {
+                    for(int i = 0; i < data(channel, baseline).cols(); i++)
+                    {
+                        metadata.avg_data.get()(baseline, i) += data(channel, baseline)(i);
+                    }
+                }
+            }
```

Comment:
Mmmmm... same comment about code reuse here, this looks awfully familiar. It's not clear to me whether this is intentionally duplicated because the previous was the reference implementation, or whether it makes sense to reuse code.

Reply:
I think I updated this to use Eigen::VectorXcd after getting annoyed with the runtime errors in casacore. I can change back and correct the old implementation if you think there is a need to optimize this.

Reply:
No problem with it being optimized or not, I was just concerned with the code being the same as `cpu::PhaseRotate`, but if I understand correctly this is another case of this file being the reference implementation.